lib/spack/spack/config.py

```diff
@@ -608,6 +613,26 @@ def _mark_overrides(data):
         return data
 
 
+def _mark_internal(data, name):
+    """Add a simple name mark to raw YAML/JSON data.
+
+    This is used by `spack config blame` to show where config lins came from.
+    """
+    if isinstance(data, dict):
+        d = syaml.syaml_dict((_mark_internal(k, name), _mark_internal(v, name))
+                             for k, v in data.items())
+    elif isinstance(data, list):
+        d = syaml.syaml_list(_mark_internal(e, name) for e in data)
+    else:
+        d = syaml.syaml_type(data)
+
+    if syaml.markable(d):
+        d._start_mark = yaml.Mark(name, None, None, None, None, None)
+        d._end_mark = yaml.Mark(name, None, None, None, None, None)
```

Comment:
Maybe a keyword binding here would be more readable:
```python
d._end_mark = yaml.Mark(name=name, index=None, line=None, column=None, buffer=None, pointer=None)
```
?

Reply:
I'm trying to stick with the way the original YAML code works.  While I agree that it could be more readable, I figure that if they don't rely on names of parameters, we shouldn't either.  The YAML code is pretty nasty.